tests/test_blockchain_gateway.py

```diff
@@ -22,21 +23,48 @@ def ipfs_client(config_manager):
     return ipfsapi.connect(config.get('BLOCKCHAIN', 'host'), 
                             config.getint('BLOCKCHAIN', 'ipfs_port'))
 
-@pytest.fixture
+@pytest.fixture(scope='session')
 def communication_manager():		
     class MockCommunicationManager:
         def __init__(self):
-            self.dummy1 = None
-            self.dummy2 = None
-        def inform(self, dummy1, dummy2):
-            self.dummy1 = dummy1
-            self.dummy2 = dummy2
+            self.dummy_msg_type = "None"
+            self.data_provider_info = "None"
+            self.job_info = "None"
+        def inform(self, dummy1, payload):
+            self.dummy_msg_type = dummy1
+            self._create_session(payload)
+        def _create_session(self, payload):
+            assert payload.get(TxEnum.KEY.name) is MessageEventTypes.NEW_SESSION.name, \
+                "Expected a new session but got {}".format(payload.get(TxEnum.KEY.name))
+            initialization_payload = payload.get(TxEnum.CONTENT.name)
+            self._setup_optimizer(initialization_payload)
+        def _setup_optimizer(self, initialization_payload):
+            self.data_provider_info = initialization_payload.get(TxEnum.KEY.name)
+            self.job_info = initialization_payload.get(TxEnum.CONTENT.name)
+            serialized_job = self.job_info.get('serialized_job')
+            self.job = deserialize_job(serialized_job)
+            self.job.uuid = self.data_provider_info.get("dataset_uuid")
+            self.job.label_column_name = self.data_provider_info.get("label_column_name")
+            assert self.job.uuid, "uuid of job not set!"
+        def reset(self):
+            self.dummy_msg_type = "None"
+            self.data_provider_info = "None"
+            self.job_info = "None"
     return MockCommunicationManager()
 
-@pytest.fixture	
-def blockchain_gateway(config_manager, communication_manager, ipfs_client):	
+@pytest.fixture(scope='session')
+def dataset_manager():
+    class MockDatasetManager:
+        def __init__(self):
+            self.mappings = [1234]
```

Comment:
Nitpick but mappings is technically a dictionary.

Reply:
M O C K D A T A S E T M A N A G E R

Reply:
still though, it's misleading

Reply:
fixed